src/test/java/com/exasol/adapter/dialects/exasol/AbstractExasolSqlDialectIT.java

```diff
@@ -743,6 +752,63 @@ void testSelectStarConvertedToColumnsListNestedJoinReversed() {
                         .matches());
     }
 
+    protected com.exasol.adapter.dialects.exasol.DataTypeAssertion.Builder typeAssertionFor(final String columnType) {
+        return DataTypeAssertion.builder(this).withColumnType(columnType);
+    }
+
+    void assertVirtualSchemaTypes(final DataTypeAssertion assertion) {
+        final Table table = createSingleColumnTable(assertion.getColumnType()).insert(assertion.getValue());
+        final VirtualSchema virtualSchema = createVirtualSchema(this.sourceSchema);
+        try {
+            assertAll( //
+                    () -> assertTypeofColumn(virtualSchema, table, assertion.getExpectedTypeOf()),
+                    () -> assertDescribeColumnType(virtualSchema, table, assertion.getExpectedDescribeType()),
+                    () -> assertResultSetType(virtualSchema, table, assertion.getExpectedResultSetType()));
```

Comment:
For the first and third you could also you https://github.com/exasol/hamcrest-resultset-matcher/

Reply:
👍